src/Functions/formatDateTime.cpp

```diff
@@ -855,289 +914,445 @@ class FunctionFormatDateTimeImpl : public IFunction
             }
             *pos++ = '\0';
 
-            dst_offsets[i] = pos - begin;
+            res_offsets[i] = pos - begin;
         }
 
-        dst_data.resize(pos - begin);
+        res_data.resize(pos - begin);
         return col_res;
     }
 
     template <typename T>
-    size_t parseFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32 scale, String & out_template) const
+    size_t parseFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32 scale, bool only_fixed_width_formatters, String & out_template) const
     {
-        static_assert(
-            format_syntax == FormatDateTimeTraits::FormatSyntax::MySQL || format_syntax == FormatDateTimeTraits::FormatSyntax::Joda,
-            "format syntax must be one of MySQL or Joda");
+        static_assert(format_syntax == FormatSyntax::MySQL || format_syntax == FormatSyntax::Joda);
 
-        if constexpr (format_syntax == FormatDateTimeTraits::FormatSyntax::MySQL)
-            return parseMySQLFormat(format, instructions, scale, out_template);
+        if constexpr (format_syntax == FormatSyntax::MySQL)
+            return parseMySQLFormat(format, instructions, scale, only_fixed_width_formatters, out_template);
         else
-            return parseJodaFormat(format, instructions, scale, out_template);
+            return parseJodaFormat(format, instructions, scale, only_fixed_width_formatters, out_template);
     }
 
     template <typename T>
-    size_t parseMySQLFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32 scale, String & out_template) const
+    size_t parseMySQLFormat(const String & format, std::vector<Instruction<T>> & instructions, UInt32 scale, bool only_fixed_width_formatters, String & out_template) const
     {
         auto add_extra_shift = [&](size_t amount)
         {
             if (instructions.empty())
-                instructions.emplace_back(&Instruction<T>::mysqlNoop);
+            {
+                Instruction<T> instruction;
+                instruction.setMysqlFunc(&Instruction<T>::mysqlNoop);
+                instructions.push_back(instruction);
+            }
             instructions.back().extra_shift += amount;
         };
 
-        auto add_instruction_or_extra_shift = [&](auto * func [[maybe_unused]], size_t amount [[maybe_unused]])
+        auto add_literal_instruction = [&](std::string_view literal)
+        {
+            Instruction<T> instruction;
+            instruction.setMysqlFunc(&Instruction<T>::mysqlLiteral);
+            instruction.setLiteral(literal);
+            instructions.push_back(instruction);
```

Comment:
```suggestion
            instructions.push_back(std::move(instruction));
```

same for all `push_back(instruction)`

Reply:
Yepp, done.